intera_examples/scripts/constrained_zeroG.py

```diff
@@ -135,134 +290,14 @@ def main():
-            unconstrained_axes = zero_stiffness_axes
-
-        # if the axes are specified by individual options
-        if (sum_mode_list==0 and sum_zero_stiffness_axes==0 and sum_free_axis_list>0):
-            unconstrained_axes = free_axis_list
-
-        # set the stiffness to zero by default
-        interaction_options.set_K_impedance([0, 0, 0, 0, 0, 0])
-
-        # set the axes with maximum stiffness
-        interaction_options.set_max_impedance(boolToggle(int2bool(unconstrained_axes)))
-
-        interaction_options.set_in_endpoint_frame(args.in_endpoint_frame)
 
-        # set nullspace stiffness to zero if nullspace_only option is provided
-        if args.nullspace_only:
-            K_nullspace = [0, 0, 0, 0, 0, 0, 0]
-        else:
-            K_nullspace = args.K_nullspace
-
-        interaction_options.set_K_nullspace(K_nullspace)
-
-        if len(args.interaction_frame) < 7:
-            rospy.logerr('The number of elements must be 7!')
-        elif len(args.interaction_frame) == 7:
-            quat_sum_square = args.interaction_frame[3]*args.interaction_frame[3] + args.interaction_frame[4]*args.interaction_frame[4]
-            + args.interaction_frame[5]*args.interaction_frame[5] + args.interaction_frame[6]*args.interaction_frame[6]
-            if quat_sum_square  < 1.0 + 1e-7 and quat_sum_square > 1.0 - 1e-7:
-                interaction_frame = Pose()
-                interaction_frame.position.x = args.interaction_frame[0]
-                interaction_frame.position.y = args.interaction_frame[1]
-                interaction_frame.position.z = args.interaction_frame[2]
-                interaction_frame.orientation.w = args.interaction_frame[3]
-                interaction_frame.orientation.x = args.interaction_frame[4]
-                interaction_frame.orientation.y = args.interaction_frame[5]
-                interaction_frame.orientation.z = args.interaction_frame[6]
-                interaction_options.set_interaction_frame(interaction_frame)
-            else:
-                rospy.logerr('Invalid input to quaternion! The quaternion must be a unit quaternion!')
-        else:
-            rospy.logerr('Invalid input to interaction_frame!')
-
-        # always enable the rotations for constrained zero-G
-        interaction_options.set_rotations_for_constrained_zeroG(True)
-
-        msg = interaction_options.to_msg()
-
-        # print the resultant interaction options once
-        rospy.loginfo(msg)
-        pub.publish(msg)
-
-        rs = intera_interface.RobotEnable(CHECK_VERSION)
-        if args.rate > 0:
-            while not rospy.is_shutdown() and rs.state().enabled:
-                rate.sleep()
-                pub.publish(msg)
-
-    except rospy.ROSInterruptException:
-        rospy.logerr('Keyboard interrupt detected from the user. %s',
-                     'Exiting the node...')
+    args = parser.parse_args(rospy.myargv()[1:])
```

Comment:
Why not just use:  `args = parser.parse_args()` ?

Reply:
This is the [wiki-recommended method to deal with rospy args](http://wiki.ros.org/rospy/Overview/Initialization%20and%20Shutdown#Accessing_your_command-line_arguments):
```
# Return a copy of sys.argv with remapping arguments removed. 
# You can optionally pass in your own argv array to have that filtered instead. 
rospy.myargv(argv=sys.argv)
```
Though I'm not sure what the `myargv` bit is doing...